infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -95,6 +106,53 @@
     private CorfuMsgHandler handler = new CorfuMsgHandler()
             .generateHandlers(MethodHandles.lookup(), this);
 
+    private final ConcurrentHashMap<UUID, Long> trimMap;
+
+    /**
+     * This cache services requests for data at various addresses. In a memory implementation,
+     * it is not backed by anything, but in a disk implementation it is backed by persistent storage.
+     */
+    private final LoadingCache<LogAddress, LogData> dataCache;
+    private final long maxCacheSize;
+
+    private final StreamLog streamLog;
+
+    private final BatchWriter<LogAddress, LogData> batchWriter;
+
+    public LogUnitServer(ServerContext serverContext) {
+        this.opts = serverContext.getServerConfig();
+        this.serverContext = serverContext;
+        maxCacheSize = Utils.parseLong(opts.get("--max-cache"));
+
+        if ((Boolean) opts.get("--memory")) {
+            log.warn("Log unit opened in-memory mode (Maximum size={}). " +
+                    "This should be run for testing purposes only. " +
+                    "If you exceed the maximum size of the unit, old entries will be AUTOMATICALLY trimmed. " +
+                    "The unit WILL LOSE ALL DATA if it exits.", Utils.convertToByteStringRepresentation(maxCacheSize));
+            streamLog = new InMemoryStreamLog();
+        } else {
+            String logdir = opts.get("--log-path") + File.separator + "log";
+            File dir = new File(logdir);
+            if (!dir.exists()) {
+                dir.mkdir();
+            }
+            streamLog = new StreamLogFiles(logdir, (Boolean) opts.get("--no-verify"));
+        }
+
+        batchWriter = new BatchWriter(streamLog);
+
+        dataCache = Caffeine.<LogAddress, LogData>newBuilder()
+                .<LogAddress, LogData>weigher((k, v) -> v.getData() == null ? 1 : v.getData().readableBytes())
```

Comment:
Or if you rearranged the construction,

```java
dataCache = Caffeine.newBuilder()
    .writer(batchWriter)
    .weigher((k, v) -> v.getData() == null ? 1 : v.getData().readableBytes())
    .build(this::handleRetrieval);
```
The lambda is inferred with the types instead of the raw `Object`.